src/background/telemetry/chainned-telemetry-client.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { TelemetryClient } from 'background/telemetry/telemetry-client';
+
+export class ChainnedTelemetryClient implements TelemetryClient {
```

Comment:
I like the idea of a multiplexing telemetry client, but I think maybe the data structure would be more straightforward and the most complex bit of new code could be removed by doing something like this instead of the linked list:

```ts
export class MultiplexingTelemetryClient implements TelemetryClient {
    constructor(private readonly wrappedClients: TelemetryClient[]);

    public enableTelemetry(): void {
        this.wrappedClients.forEach(c => c.enableTelemetry());
    }

    public disableTelemetry(): void {
        this.wrappedClients.forEach(c => c.disableTelemetry());
    }

    public trackEvent(name: string, properties?: Object): void {
        this.wrappedClients.forEach(c => c.trackEvent(name, properties));
    }
}
```

Reply:
Like it. Will update my PR with this changes in a little bit.

Reply:
Updated.